### ISSUE TITLE
About training and testing the code.

### ISSUE THREAD

**Issue by UserXiangYang** [heitorrapela/MiPa#2]
Could you please provide the Settings of the hyperparameters for training and testing on the dataset in the paper?  e.g., python train -c ........   Thinks!

**Comment by qiherong** [comment]
> Could you please provide the Settings of the hyperparameters for training and testing on the dataset in the paper? e.g., python train -c ........ Thinks!

+1

**Comment by maengjemo** [comment]
+1

**Comment by heitorrapela** [comment]
Hello, thanks for the interest in our work, sorry for the delay, I have been a little busy with the end of my phd, 

These are some commands that you can use for training a model and eval, here is an example for p = 0.25 (fixed sampling), for the experiments related to 25% of IR patches and 75% RGB, the model is DINO 4scale with swin backbone with seed 43, the additional flags are related to original fine-tuning strategy for DINO, also --dataset_file llvip to select the llvip dataset:

## Dataset preparation
Create a symbolic link for the datasets and place it inside: datasets/ (It should be inside src: MiPa/src/datasets/)
For example: ./datasets/LLVIP
Inside the LLVIP folder, place the .json coco format labels that I am providing.

Annotations for RGB and IR splits used in the main results of the paper:

[llvip_ir_test.json](https://github.com/user-attachments/files/21208397/llvip_ir_test.json)
[llvip_ir_train.json](https://github.com/user-attachments/files/21208395/llvip_ir_train.json)
[llvip_ir_valid.json](https://github.com/user-attachments/files/21208396/llvip_ir_valid.json)
[llvip_rgb_test.json](https://github.com/user-attachments/files/21208398/llvip_rgb_test.json)
[llvip_rgb_train.json](https://github.com/user-attachments/files/21208400/llvip_rgb_train.json)
[llvip_rgb_valid.json](https://github.com/user-attachments/files/21208399/llvip_rgb_valid.json)


## Pre-trained checkpoints 
For the pre-trained checkpoint used: https://github.com/IDEA-Research/DINO
Directly link for it: https://drive.google.com/drive/folders/1qD5m1NmK0kjE5hh-G17XUX751WsEG-h_?usp=sharing
Add it inside ./weights/

## Run the script


Go inside to run the script: cd MiPa/src/DINO/

## TRAIN

python main.py --output_dir logs/DINO/llvip/DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43/ -c config/DINO/DINO_llvip_4scale_swin.py --dataset_file llvip --modality both --options dn_scalar=100 embed_init_tgt=TRUE dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False dn_box_noise_scale=1.0 --pretrain_model_path ./weights/checkpoint0029_4scale_swin.pth --finetune_ignore label_enc.weight class_embed --sampling_method fixed --ir_modality_ratio 0.25 --seed 43 > DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43.txt

## EVAL RGB
python main.py --output_dir logs/DINO/llvip/DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43_eval_rgb/ -c config/DINO/DINO_llvip_4scale_swin.py --dataset_file llvip --modality rgb --eval --resume ./logs/DINO/llvip/DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43/checkpoint_best_regular.pth --eval-set test --options dn_scalar=100 embed_init_tgt=TRUE dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False dn_box_noise_scale=1.0 > DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43_eval_rgb.txt

## EVAL IR
python main.py --output_dir logs/DINO/llvip/DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43_eval_ir/ -c config/DINO/DINO_llvip_4scale_swin.py --dataset_file llvip --modality ir --eval --resume ./logs/DINO/llvip/DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43/checkpoint_best_regular.pth --eval-set test --options dn_scalar=100 embed_init_tgt=TRUE dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False dn_box_noise_scale=1.0 > DINO_llvip_4scale_swin_both_fixed_p_0.25_s_43_eval_ir.txt

**Comment by heitorrapela** [comment]
Additionally, we are planning to improve the documentation when I have a bit more time, and I will try to clean the instructions for other kinds of runs and provide some checkpoints for just evaluations (for this, I need to check the best place to host the models due to the size of some models)
